src/components/Store.tsx

```diff
@@ -115,12 +115,16 @@ function reducer(state: ChatState, action: A): ChatState {
                       },
                   }
         case ActionType.Reset:
+            const client = new StreamrClient({
+                auth: { ethereum: state.ethereumProvider as any },
+            })
```

Comment:
Wrong spot. Let's do it in `ActionType.SetEthereumProvider` and here let's just do
```ts
return {
    ...getInitialChatState(),
    // Keep the provider. We may wanna extract it from "chat" state. It's actually more
    // of a browser/app state piece.
    ethereumProvider: state.ethereumProvider,
    ethereumProviderReady: state.ethereumProviderReady,
    metamaskStreamrClient: state.metamaskStreamrClient, // HERE
}
```

Reply:
This would create a metamask client on logout. ;)